base/applications/cmdutils/more/more.c

```diff
@@ -134,28 +312,103 @@ PagePrompt(PCON_PAGER Pager, DWORD Done, DWORD Total)
      */
     ConClearLine(Pager->Screen->Stream);
 
+    switch (s_chSubCommand)
+    {
+        case L'P':
+            s_chSubCommand = 0;
+            Pager->ScrollRows = nLines;
+            return TRUE;
+        case L'S':
+            s_chSubCommand = 0;
+            s_dwFlags |= FLAG_PLUSn;
+            s_nNextLineNo = Pager->lineno + nLines;
+            Pager->ScrollRows = Pager->ScreenRows - 1;
+            return TRUE;
+        case L'-':
+            s_chSubCommand = 0;
+            Pager->PagerAction = MorePageActionDoNothing;
+            return TRUE;
+    }
+    s_chSubCommand = 0;
+
     /* Ctrl+C or Ctrl+Esc: Control Break */
-    if ((KeyEvent.wVirtualKeyCode == VK_ESCAPE) ||
-        ((KeyEvent.wVirtualKeyCode == L'C') &&
-         (KeyEvent.dwControlKeyState & (LEFT_CTRL_PRESSED | RIGHT_CTRL_PRESSED))))
+    if (fCtrl && ((KeyEvent.wVirtualKeyCode == VK_ESCAPE) ||
+                  (KeyEvent.wVirtualKeyCode == L'C')))
     {
         /* We break, output a newline */
         WCHAR ch = L'\n';
         ConStreamWrite(Pager->Screen->Stream, &ch, 1);
         return FALSE;
     }
 
-    /* 'Q': Quit */
-    // FIXME: Available only when command extensions are enabled.
-    if ((KeyEvent.wVirtualKeyCode == L'Q') &&
-        !(KeyEvent.dwControlKeyState & (LEFT_CTRL_PRESSED | RIGHT_CTRL_PRESSED)))
+    if (s_dwFlags & FLAG_E) /* If extended features are available */
     {
-        /* We break, output a newline */
-        WCHAR ch = L'\n';
-        ConStreamWrite(Pager->Screen->Stream, &ch, 1);
-        return FALSE;
+        /* 'Q': Quit */
+        if ((KeyEvent.wVirtualKeyCode == L'Q') && !fCtrl)
+        {
+            /* We break, output a newline */
+            WCHAR ch = L'\n';
+            ConStreamWrite(Pager->Screen->Stream, &ch, 1);
+            return FALSE;
+        }
+
+        /* 'F': Next file */
+        if ((KeyEvent.wVirtualKeyCode == L'F') && !fCtrl)
+        {
+            Pager->PagerAction = MorePageActionNextFile;
+            return TRUE;
```

Comment:
When you do: `MORE file1 file2` and you are viewing `file1` pressing "F" for going to next file should:
- stop the paging of the current file (so, `PagePrompt` should return FALSE),
- detect in the pager's caller (in main()) that paging didn't stop prematurely because we want to quit, but because we want to go to the next file
- go to the next file.

Reply:
How to detect?

Reply:
This `PagePrompt()` function could set some global (in more.c) `BOOL bDoNextFile;` variable to TRUE.
Then, in the `main()` function, at the place where the following check is done:
```c
            /* If we Ctrl-C/Ctrl-Break, stop everything */
            if (!bContinue)
            {
                CloseHandle(hFile);
                goto Quit;
            }
```
you could modify it into something like:
```c
            /* Check whether we should stop displaying this file */
            if (!bContinue)
            {
                if (bDoNextFile)
                {
                    /* Bail out and continue with the other files */
                    bDoNextFile = FALSE; // Reset the flag.
                    break;
                }

                /* We Ctrl-C/Ctrl-Break, stop everything */
                CloseHandle(hFile);
                goto Quit;
            }
```

Reply:
OK.